### PR TITLE
Better startup options

### PR DESCRIPTION
**Pull request by halgari**
Makes several changs to the CLI system:

* You can now prefix a CLI command with `as-main` to have the CLI process try to become main. So `NexusMods.App.exe as-main list-loadouts` will list the loadouts and assume the app is not otherwise running. No UI will be started in this mode
* Same as the above, you can prefix `as-main-ui` to also start the UI which will run parallel with the CLI command
* If no prefix is given and no main is running the CLI now tries to start the main app in a different process
* Due to all these new modes and options, abstracted the CLI parsing behind a `StartupOptions` class that contains all this info. We can now pass that around instead of `isMain`
* Reworked several parts of the startup code to be happy with these changes
* Cleaned up protocol-invoke a ton, it was using a strange message queue system from when we still had multi-process support. Now it just uses a `Subject`
* Added a method to the OSInterop code to get the path of the running executable. 

This all works on Linux and should work on Windows as well. Something seems broken on my box. It works from the commandline for me in windows if I call `protocol-invoke` from the CLI, but windows isn't loading the app for some reason. Perhaps someone reviewing this PR with windows access can try to log in via the app and see what they get? I've blown about 3 hours debugging this and am clueless as to what's wrong.

### CODE REVIEW

**Comment by halgari** [comment]
Fixes #1360

**Review by erri120 — CHANGES_REQUESTED** [review]

**Comment by erri120** [comment]
@Al12rs do you want to test this on Windows? Just open the App and try to log in. It works on Linux but apparently has some issues on Windows.

**Comment by Al12rs** [comment]
> @Al12rs do you want to test this on Windows? Just open the App and try to log in. It works on Linux but apparently has some issues on Windows.

Yeah, I have been debugging App not showing up when starting a nxm download. I think the problem is that the original process closes, and as the parent is dies, the child process is also killed, so main process is also closed.

**Review by Al12rs — CHANGES_REQUESTED** [review]
So the problem is that in Windows, child processes are not automatically terminated when the parent process ends. However, if the child process is reading from standard input (stdin) or writing to standard output (stdout), it will receive an end-of-file (EOF) or broken pipe signal and may exit as a result.

To avoid this from happening we can simply make the App not be a console application by changing `<OutputType>Exe</OutputType>` to `<OutputType>WinExe</OutputType>`. 

This will remove the console from showing up in windows when starting the app, and as a happy side effect, will prevent the main app from closing if the parent process console is closed. 

This took me a while to debug and figure out as well

File: src/NexusMods.App/Program.cs [hunk 1]
```diff
@@ -88,9 +100,61 @@ public static int Main(string[] args)
...
+            return 0;
+        }
+        catch (NoMainProcessStarted _)
+        {
+            var interop = services.GetRequiredService<IOSInterop>();
+            var ownExe = interop.GetOwnExe();
+            _logger.LogInformation("No main process started, starting {OwnExe}", ownExe);
+            var processInfo = new ProcessStartInfo
+            {
+                FileName = ownExe.ToString(),
```
[the remark this answers is not in the document]
Comment (halgari) [reply src/NexusMods.App/Program.cs:119]: According to StackOverflow I'm pretty sure it does, also I've never escaped these in the past and it's worked, so I think we're good.

**Review by erri120 — APPROVED** [review]

**Comment by halgari** [comment]
> So the problem is that in Windows, child processes are not automatically terminated when the parent process ends. However, if the child process is reading from standard input (stdin) or writing to standard output (stdout), it will receive an end-of-file (EOF) or broken pipe signal and may exit as a result.
> 
> To avoid this from happening we can simply make the App not be a console application by changing `<OutputType>Exe</OutputType>` to `<OutputType>WinExe</OutputType>`.
> 
> This will remove the console from showing up in windows when starting the app, and as a happy side effect, will prevent the main app from closing if the parent process console is closed.
> 
> This took me a while to debug and figure out as well

I set `UseShellExecute = true` to fix this in the latest version which I think works, can you give it a try?

**Pull request merged**